src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -241,14 +261,15 @@ private IServiceCollection BuildCommonServices(out AggregateException hostingSta
             var appEnvironment = PlatformServices.Default.Application;
             var contentRootPath = ResolveContentRootPath(_options.ContentRootPath, appEnvironment.ApplicationBasePath);
             var applicationName = _options.ApplicationName ?? appEnvironment.ApplicationName;
-            var hostingContext = new WebHostBuilderContext
+
+            _hostingContext = new WebHostBuilderContext
```

Comment:
_hostingContext should be set in the initial constructor so Context never returns null. You have the initial _config at that point too.